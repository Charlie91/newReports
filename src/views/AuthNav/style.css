.auth-nav {
  margin-bottom: 60px; }
  .auth-nav a {
    font-family: ProximaNova, sans-serif;
    font-size: 1.625em;
    font-weight: 600;
    color: #354052;
    display: inline-block;
    line-height: 40px; }
    .auth-nav a:hover {
      text-decoration: none; }
    .auth-nav a.active {
      border-bottom: solid 5px rgba(142, 214, 250, 0.8); }

.auth-btn {
  width: 100%;
  background: #8ed6fa;
  color: #fff;
  font-size: 130%;
  border-radius: 3px; }
  .auth-btn:disabled, .auth-btn.dsbl {
    background-color: rgba(216, 216, 216, 0.6);
    box-shadow: 0 0 2px #b3b3b3; }

button.btn.back-btn {
  background: no-repeat;
  color: rgba(57, 67, 85, 0.8);
  font-size: 18px; }
  button.btn.back-btn:focus {
    box-shadow: none; }

.auth-window {
  width: 40%;
  margin: auto;
  background-color: rgba(255, 255, 255, 0.55);
  text-align: center;
  padding: 60px 80px;
  border-radius: 3px;
  -webkit-backdrop-filter: blur(8.5px);
  box-shadow: 5px 8px 15px 0 rgba(149, 149, 149, 0.15), 0 -1px 15px 0 rgba(194, 194, 194, 0.07); }
  .auth-window label {
    font-weight: 400;
    width: 70%;
    position: relative;
    margin-bottom: 35px; }
  .auth-window form {
    text-align: center; }
  .auth-window input:-webkit-autofill, .auth-window input:-webkit-autofill:focus {
    -webkit-text-fill-color: #3e515b !important;
    color: #3e515b !important;
    -webkit-box-shadow: 0 0 0px 1000px #f5f7f8 inset; }
  .auth-window input {
    background: none;
    border: none;
    border-bottom: solid 1px rgba(161, 161, 161, 0.8);
    width: 100%;
    padding-left: 0;
    display: block;
    margin: auto;
    margin-bottom: 5px;
    font-size: 110%; }
    .auth-window input:focus {
      background: transparent;
      border: none;
      border-bottom: 1px solid #8bc8e8;
      box-shadow: none; }
  .auth-window .hintMessage {
    width: 70%;
    margin: auto; }

.label-hint {
  position: absolute;
  left: 0px;
  top: -20px;
  color: #c1cacb; }

@media screen and (max-width: 1600px) {
  .auth-window {
    width: 50% !important; } }

@media screen and (max-width: 1400px) {
  .auth-window {
    width: 60% !important; } }

@media screen and (max-width: 991px) {
  .auth-window {
    width: 60% !important; }
    .auth-window label {
      width: 100%; }
    .auth-window .hintMessage {
      width: 100%; }
  .auth-nav a {
    font-size: 2.3vw; }
    .auth-nav a:active {
      box-shadow: 0 2px 2px #b9b9b9;
      background: #d8dedc;
      padding: 1px;
      -webkit-tap-highlight-color: transparent; } }

@media screen and (max-width: 767px) {
  .auth-window {
    width: 100% !important;
    border-radius: 10px;
    padding: 60px 20px; }
    .auth-window label {
      width: 100% !important; }
  .auth-nav a {
    font-size: 1.5em; }
  .dropdown__content.dropdown-content {
    height: auto;
    max-height: 70vh; }
  .empty-tmpl-app .container-fluid {
    padding: 24px; } }

@media screen and (max-width: 400px) {
  .empty-tmpl-app .container-fluid {
    padding: 20px; }
  .auth-nav {
    margin-bottom: 60px;
    font-size: 90%;
    padding: 10px; } }

@media screen and (max-width: 350px) {
  .auth_links {
    padding: 0; } }
