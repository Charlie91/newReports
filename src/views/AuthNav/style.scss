.auth-nav{
  margin-bottom: 53px;
  a {
    width:209px;
    font-family: ProximaNova, sans-serif;
    font-size: 1.625em;
    font-weight: 500;
    color: #354052;
    display: inline-block;
    line-height: 41px;
    padding : 0 7px 0 7px;
    &:hover {
      text-decoration: none
    }
    &.active {
      border-bottom: solid 5px rgba(142, 214, 250, 0.8)
    }
  }
}

.clear-field {
  background-image:url('/img/clear.png');
  background-size:16px;
  background-position: right;
  background-color: transparent !important;
  background-repeat: no-repeat;
}

.nextStep .auth-btn{
  width:185px
}

.auth-btn{
  width:100%;
  background:#8ed6fa;
  color:#fff;
  font-size: 1.5em;
  margin-bottom: 35px;
  border-radius: 3px;
  &:disabled, &.dsbl {
    background-color: rgba(216, 216, 216, 0.6);
    box-shadow: 0 0 2px #b3b3b3;
  }
}

button.btn.back-btn {
  &:focus{
    box-shadow: none;
  }
  background: no-repeat;
  color: rgba(57, 67, 85, 0.8);
  font-size: 1.5em;
}

.auth-window {
  width: 805px;
  //width:40%;
  margin: auto;
  background-color: rgba(255, 255, 255, 0.55);
  text-align: center;
  padding:63px 80px 0px 80px;
  border-radius: 10px;
  -webkit-backdrop-filter: blur(8.5px);
  box-shadow: 5px 8px 15px 0 rgba(149, 149, 149, 0.15), 0 -1px 15px 0 rgba(194, 194, 194, 0.07);

  label,.nextStep{
    font-weight:400;
    width:70%;
    position:relative;
    margin-bottom: 70px;
  }
  form {
    text-align: center;
  }
  input:-webkit-autofill,input:-webkit-autofill:focus {
    -webkit-text-fill-color: #3e515b !important;
    color: #3e515b !important;
    -webkit-box-shadow: 0 0 0px 1000px #fafbfc inset;
  }

  input {
    color: #363636;
    background: none;
    border: none;
    border-bottom: solid 1px rgba(161, 161, 161, 0.8);
    width: 100%;
    padding-left:0;
    display: block;
    margin: auto;
    margin-bottom: 2px;
    font-size: 1.5em;
    padding-right: 25px;

    &::placeholder {
      color: #363636;
    }
    &:focus {
      background: transparent;
      border:none;
      border-bottom: 1px solid #8bc8e8;
      box-shadow: none;
    }
  }
  .form-group {
    margin-bottom: 17px;
  }
  .hintMessage{
    width:70%;
    margin:auto;
    background-color: transparent;
    border: 0px;
    padding-top: 0px;
    color: #fd0101;
  }
}

.label-hint{
  position: absolute;
  left: 0px;
  top: -23px;
  color: #c1cacb;
  font-size: 1.25em;
}


@media screen and (max-width:820px){
  .auth-window {
    width: 100%;
    //width: 805px;
    border-radius:10px;

    label, .nextStep{
      width:90%!important
    }
  }
  .auth-nav {
    a {
      font-size: 1.5em;
    }
  }
  .dropdown__content.dropdown-content{
    height: auto;
    max-height: 70vh;
  }
  .empty-tmpl-app .container-fluid{
    padding:24px
  };
}

@media screen and (max-width:670px){
  .auth-window {
    padding: 0px 0px 35px 0px;
    input {
      font-size: 1.125em;
    }
    label,.nextStep {
      margin-bottom: 37px;
    }
    .form-group {
      margin-bottom: 0px;
    }
    .auth_links{
      padding: 0px;
    }
  }
  .empty-tmpl-app .container-fluid{
    padding:25px
  };
  .auth-nav {
    a {
      width: 141px;
      font-size: 1.125em !important;
      padding: 0 3px 0 3px;
    }
    margin-bottom: 27px;
    padding: 19px;
  }
  .nextStep .auth-btn{
    width:154px;
  }
  .auth-btn {
    font-size: 1.25em !important;
    width:154px;
    margin-top: -9px;
    margin-bottom: 0px;
  }
  .label-hint{
    font-size: 0.875em;
    top: -10px;
  }
  button {
    .auth-btn {
      font-size: 1.125em !important;
    }
    .btn.back-btn {
      font-size: 1.125em !important;
    }
  }
}

@media screen and (max-width:325px){
  .empty-tmpl-app .container-fluid{
    padding: 5px;
  }
}





