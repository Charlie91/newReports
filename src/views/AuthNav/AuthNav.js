import React, { Component } from 'react';
import './style.scss';
import {NavLink} from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.css';

import {
    Row,
    Col,
    Container
} from "reactstrap";

const AuthNav = (props) => {
    return (
            <Row className="auth-nav">
                <Col xs="0" sm="1" md="1"></Col>
                <Col xs="6" sm="5" md="5">
                    <NavLink to={'/authorization'} activeClassName="active">Войти</NavLink>
                </Col>
                <Col xs="6" sm="5" md="5">
                    <NavLink to={'/registration'}  activeClassName="active">Регистрация</NavLink>
                </Col>
                <Col xs="0" sm="1" md="1"></Col>
            </Row>
    )
}

export default AuthNav;