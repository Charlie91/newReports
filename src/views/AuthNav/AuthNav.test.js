import fetch from 'node-fetch';
import AuthNav from './AuthNav';
import renderer from 'react-test-renderer';
import React from 'react';
import { NavLink } from 'react-router-dom';


jest.mock('react-router-dom');
NavLink.mockImplementation(() => <div>Link</div>);

describe('Auth navigation works correctly', () => {//тестирование класса
    it('AuthNav component renders correctly', () => {
        const tree = renderer
            .create(<AuthNav/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});