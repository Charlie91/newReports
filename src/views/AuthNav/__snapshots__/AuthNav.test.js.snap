// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth navigation works correctly AuthNav component renders correctly 1`] = `
<div
  className="auth-nav row"
>
  <div
    className="col-0 col-sm-1 col-md-1"
  >
     
  </div>
  <div
    className="auth_links col-6 col-sm-5 col-md-5"
  >
    <div>
      Link
    </div>
  </div>
  <div
    className="auth_links col-6 col-sm-5 col-md-5"
  >
    <div>
      Link
    </div>
  </div>
  <div
    className="col-0 col-sm-1 col-md-1"
  />
</div>
`;
